### PR TITLE
feat(monitoring): docker-compose.yml runs Gossamer + Prometheus + Grafana

### PR DESCRIPTION
**Pull request by qdm12**
## Changes

TLDR: `docker-compose up` then http://localhost:3000 to see a nice dashboard.

- Tailor docker-compose.yml for local development and metrics viewing
- Add Gossamer to `docker-compose.yml`
- Add Grafana to `docker-compose.yml`
- Move prometheus.yml to `docker/prometheus/` (so bind mount works on OSX, Windows)
- Add Grafana configurations and provisioning to `docker/grafana`
- Remove unneeded Prometheus commands
- Do not map Prometheus server port, only expose to Grafana
- Add `/docker` to `.dockerignore`
- Set docker-compose.yml version to `'3'` for greater compatibility
- `docker-compose build` builds Gossamer image
- Add commented lines to persist Prometheus and Grafana data
- Update using Prometheus document page

Side notes:

- I could not find a working substrate dashboard I could plug in grafana with gossamer
- `gossamer_network_node_latency_ms` is always 0 while syncing 🤔 
- `gossamer_state_transaction_pool_total` is always 0 while syncing 🤔 
- `gossamer_state_transaction_queue_total` is always 0 while syncing 🤔 
- `gossamer_grandpa_round` is always 0  while syncing🤔 
- `gossamer_block_leaves_total` is always 1  while syncing🤔 

## Tests

```
git checkout qdm12/prometheus-grafana
docker-compose up
```

Then go to http://localhost:3000

image

## Issues

#2693 

## Primary Reviewer

@EclesioMeloJunior

### CODE REVIEW

**Review by EclesioMeloJunior — APPROVED** [review]
LGTM!

**Review by noot — APPROVED** [review]
dashboard works for me, looks good :D
